scripts/embed_ui_assets.sh

```diff
@@ -27,15 +27,11 @@ if [ ! -d "ui/build" ]; then
   exit 1
 fi
 
-if [ "${NO_ASSET_BUILD_TAG:-}" = "1" ]; then
-  BUILD_TAG_PARAMETER=""
-else
-  BUILD_TAG_PARAMETER="-tags ${ASSET_BUILD_TAG:-ui_server}"
-fi
-
 echo "+ Embed UI assets"
 
-go-bindata-assetfs -pkg uiserver -prefix ui $BUILD_TAG_PARAMETER ui/build/...
+go build -o bin/assets_generate tools/assets_generate/main.go && bin/assets_generate
```

Comment:
How about using [go-generate](https://github.com/shurcooL/vfsgen#go-generate-usage)?

Reply:
Hi, I'm not quite familiar with  `go generate`, I'm wondering what's the benefit of using it here? 
Here is what I got after reading the doc:
1. Create a go file  including `//go:generate go run  assets_generate.go` comment
2. Export `BUILD_TAG_PARAMETER` in `embed_ui_assets.sh`
3. Fill `BuildTags` option of `vfsgen.Options` using `os.Getenv("BUILD_TAG_PARAMETER")`
4. Call `go generate` in `embed_ui_assets.sh`

Am I missing something?

Reply:
Yes, looks good